service/collector/ec2instances.go

```diff
@@ -0,0 +1,241 @@
+package collector
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	clientaws "github.com/giantswarm/aws-operator/client/aws"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	// labelInstance is the metric's label key that will hold the ec2 instance ID.
+	labelInstance = "ec2instance"
+
+	// labelInstanceState is a label that will contain the instance state string
+	labelInstanceState = "state"
+
+	// labelInstanceStatus is a label that will contain the instance status check value
+	labelInstanceStatus = "status"
+
+	// subsystemEC2 will become the second part of the metric name, right after namespace.
+	subsystemEC2 = "ec2"
+)
+
+var (
+	ec2InstanceStatus = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, subsystemEC2, "status"),
```

Comment:
From a metrics design point of view there is more to EC2 statuses. When this is about instances, we should extend the metric name accordingly. 

```suggestion
		prometheus.BuildFQName(namespace, subsystemEC2, "instances_status"),
```

Reply:
Yes, fine with me. Why `instances_status` and not `instance_status`?

Reply:
I was pluralizing all the way through. The singular here would also work. Did not think anything special here.

Reply:
Changed to `instances_status`